underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AInputStream.java

```diff
@@ -105,56 +96,59 @@ public int read(byte[] b) throws IOException {
 
   @Override
   public int read(byte[] b, int offset, int length) throws IOException {
-    if (length == 0) {
-      return 0;
+    S3Object object = getObject(mPos, mPos + length);
```

Comment:
precheck param

Reply:
Will add unit test to understand the corner cases better